python/ray/experimental/state/api.py

```diff
@@ -1,195 +1,239 @@
-import requests
 import warnings
 
-from typing import List, Tuple
+from dataclasses import fields
+from typing import Dict, List, Optional, Tuple, Union
 
 import ray
+
 from ray.experimental.state.common import (
-    SupportedFilterType,
-    ListApiOptions,
-    DEFAULT_RPC_TIMEOUT,
     DEFAULT_LIMIT,
+    DEFAULT_RPC_TIMEOUT,
+    ListApiOptions,
+    SupportedFilterType,
+    StateResource,
 )
 from ray.experimental.state.exception import RayStateApiException
+from ray.dashboard.modules.dashboard_sdk import SubmissionClient
 
 
-# TODO(sang): Replace it with auto-generated methods.
-def _list(
-    resource_name: str,
-    options: ListApiOptions,
-    api_server_url: str = None,
-    _explain: bool = False,
-):
-    """Query the API server in address to list "resource_name" states.
+class StateApiClient(SubmissionClient):
+    """State API Client issues REST GET requests to the server for resource states.
 
     Args:
-        resource_name: The name of the resource. E.g., actor, task.
-        options: The options for the REST API that are translated to query strings.
-        api_server_url: The address of API server. If it is not give, it assumes the ray
-            is already connected and obtains the API server address using
-            Ray API.
-        explain: Print the API information such as API
-            latency or failed query information.
+        api_server_address: The address of API server. If it is not give, it assumes
+        the ray is already connected and obtains the API server address using Ray API.
     """
-    if api_server_url is None:
-        assert ray.is_initialized()
-        api_server_url = (
-            f"http://{ray.worker.global_worker.node.address_info['webui_url']}"
-        )
 
-    # We don't use `asdict` to avoid deepcopy.
-    # https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict
-    params = {
-        "limit": options.limit,
-        "timeout": options.timeout,
-        "filter_keys": [],
-        "filter_values": [],
-    }
-    for filter in options.filters:
-        filter_k, filter_val = filter
-        params["filter_keys"].append(filter_k)
-        params["filter_values"].append(filter_val)
-    r = requests.request(
-        "GET",
-        f"{api_server_url}/api/v0/{resource_name}",
-        params=params,
-        headers={"Content-Type": "application/json"},
-        json=None,
-        timeout=options.timeout,
-    )
-    r.raise_for_status()
+    def __init__(
+        self,
+        api_server_address: str = None,
+    ):
+        super().__init__(
+            self._get_default_api_server_address()
+            if api_server_address is None
+            else api_server_address,
+            create_cluster_if_needed=False,
```

Comment:
Do you know what it means by setting this True? (based on other clients implementation?)

Reply:
It's actually a no-ops now lollll
But I guess originally it is intended for auto cluster creation if it is not running - but I am not too sure since the code seems to be obsolete now. 
https://github.com/ray-project/ray/blob/905258dbc19753c81039f993477e7ab027960729/dashboard/modules/dashboard_sdk.py#L82-L110